src/components/main/main-layout.tsx

```diff
@@ -1,20 +1,519 @@
-import { Outlet } from 'react-router';
-import { type ReactNode } from 'react';
-import { useNavigation } from '../../hooks/use-navigation';
-
+import { Outlet, useLocation, useNavigate } from 'react-router';
+import { ReactNode, useCallback, useMemo } from 'react';
+import { AbilityModal } from '../modals/ability/ability-modal';
+import { AboutModal } from '../modals/about/about-modal';
+import type { Ancestry } from '../../models/ancestry';
+import { AncestryModal } from '../modals/ancestry/ancestry-modal';
+import type { Career } from '../../models/career';
+import { CareerModal } from '../modals/career/career-modal';
+import type { Characteristic } from '../../enums/characteristic';
+import { CharacteristicModal } from '../modals/characteristic/characteristic-modal';
+import { ClassModal } from '../modals/class/class-modal';
+import type { Complication } from '../../models/complication';
+import { ComplicationModal } from '../modals/complication/complication-modal';
+import type { Culture } from '../../models/culture';
+import { CultureModal } from '../modals/culture/culture-modal';
+import type { Domain } from '../../models/domain';
+import { DomainModal } from '../modals/domain/domain-modal';
+import type { Element } from '../../models/element';
+import { EncounterModal } from '../modals/encounter/encounter-modal';
+import type { Hero } from '../../models/hero';
+import type { HeroClass } from '../../models/class';
+import { HeroLogic } from '../../logic/hero-logic';
+import { HeroStateModal } from '../modals/hero-state/hero-state-modal';
+import type { Item } from '../../models/item';
+import { ItemModal } from '../modals/item/item-modal';
+import type { Kit } from '../../models/kit';
+import { KitModal } from '../modals/kit/kit-modal';
+import type { MonsterGroup } from '../../models/monster';
+import { MonsterGroupModal } from '../modals/monster-group/monster-group-modal';
+import { MonsterModal } from '../modals/monster/monster-modal';
+import type { Perk } from '../../models/perk';
+import { PerkModal } from '../modals/perk/perk-modal';
+import { RulesModal } from '../modals/rules/rules-modal';
+import type { Sourcebook } from '../../models/sourcebook';
+import type { SourcebookElementKind } from '../../models/sourcebook-element-kind';
+import { SourcebookLogic } from '../../logic/sourcebook-logic';
+import { SourcebooksModal } from '../modals/sourcebooks/sourcebooks-modal';
+import type { Title } from '../../models/title';
+import { TitleModal } from '../modals/title/title-modal';
+import { Utils } from '../../utils/utils';
 import pbds from '../../assets/powered-by-draw-steel.png';
+import { useNavigation } from '../../hooks/use-navigation';
+import { usePersistedHeroes } from '../../hooks/use-persisted-heroes';
+import { usePersistedPlaybook } from '../../hooks/use-persisted-playbook';
+import { usePersistedSourcebooks } from '../../hooks/use-persisted-sourcebooks';
 
 interface Props {
-	section: 'hero' | 'library' | 'encounter';
```

Comment:
this section prop didn't seem to be used for anything anyway, so i didn't keep it while rebasing to latest from your main